src/dune_engine/scheduler.ml

```diff
@@ -1181,6 +1181,7 @@ module Run = struct
     let* t = poll_init () in
     let rec loop () =
       let* _res = poll_iter t step in
+      Option.iter t.config.stats ~f:Dune_stats.flush;
```

Comment:
I suppose it may be worth pulling this out into a separate function `run_when_idle`, so it's more obvious where to add more code if needed in future.

Reply:
(We might as well document our expectations there about the kind of code that's allowed there.)